externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala

```diff
@@ -50,6 +53,39 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
   val operationManager = new SparkSQLOperationManager()
 
   private lazy val singleSparkSession = conf.get(ENGINE_SINGLE_SPARK_SESSION)
+  private lazy val shareLevel = ShareLevel.withName(conf.get(ENGINE_SHARE_LEVEL))
+
+  private lazy val userIsolatedSparkSession = conf.get(ENGINE_USER_ISOLATED_SPARK_SESSION)
+  private lazy val userIsolatedSparkSessionCache = new ConcurrentHashMap[String, SparkSession]()
```

Comment:
leaked?

Reply:
I thought about it, but can not find a better way when to release the session. It should be accepted since users can not be so many..

Reply:
I have met someone using UUID as `user`

Reply:
Use Guava Cache instead?

Reply:
ideally, It will be unexpect for user if we release the cache that users will lose their context.

yeah, we can use guava cache for some inactive timeout release. Will add a new config for that.

Reply:
> I have met someone using UUID as `user`

I know of a large cluster of a company that creates a unique UGI for each job for tracking purposes, so they have many users.

Reply:
thanks for the info !